src/main/python/run_es_regression.py

```diff
@@ -0,0 +1,236 @@
+        process = Popen(command, shell=True, stdout=PIPE)
+
+        if not echo:
+            output, err = process.communicate()
+            if process.returncode == 0: # success
+                if capture == False: return process.returncode
+                else: return output.decode('utf8').split('\n')
+            else:
+                raise RuntimeError('Error running shell comand: {}'.format(command))
+        else:
+            arr = []
+            # Modified from https://fabianlee.org/2019/09/15/python-getting-live-output-from-subprocess-using-poll/
+            while True:
+                output = process.stdout.readline()
+                if process.poll() is not None: break
+                if output:
+                    logger.info('subprocess - ' + output.decode('utf8').strip())
+                    if capture: arr.append(output.decode('utf8').strip())
+            # Make sure we read the output completely
+            while True:
+                output = process.stdout.readline()
+                if not output: break
+                logger.info('subprocess - ' + output.decode('utf8').strip())
+                if capture: arr.append(output.decode('utf8').strip())
+            if capture: return arr
+            else: return process.returncode
+
+    def is_alive(self):
+        try:
+            response = requests.get('http://localhost:9200/')
+            response.raise_for_status()
+        except: return False
+        else: return True
+
+    def does_index_exist(self, collection):
+        # Make sure ES is alive:
+        if self.is_alive():
+            try:
+                response = requests.get('http://localhost:9200/{}'.format(collection))
+                response.raise_for_status()
+            except: return False
+            else: return True
+        else: raise Exception('ES does not appear to be alive!')
+
+    def delete_index(self, collection):
+        logger.info('Deleting index {}...'.format(collection))
+        # Make sure the index exists:
+        if self.does_index_exist(collection):
+            try:
+                response = requests.request('DELETE', url='http://localhost:9200/{}'.format(collection))
+                response.raise_for_status()
+            except: return False
+            else: return True
+        else: raise Exception('The index {} does not exist!'.format(collection))
+
+    def create_index(self, collection):
+        logger.info('Creating index {}...'.format(collection))
+        # Make sure the index does not exist:
+        if not self.does_index_exist(collection):
+            filename = 'src/main/resources/elasticsearch/index-config.{}.json'.format(collection)
+            if not os.path.exists(filename):
+                raise Exception('No config found in src/main/resources/elasticsearch/ for {}!'.format(collection))
+            logger.info('Using index config for {} at {}'.format(collection, filename))
+            file = open(filename, mode='r')
```

Comment:
Python also has the `with` construct that automatically closes the file, it's the more Python-ic way...:
```
with open(filename, mode='r') as file:
    json = file.read()
```

Reply:
ah, good call!